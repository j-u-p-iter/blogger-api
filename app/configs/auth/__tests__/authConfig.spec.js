import { createAuthConfig } from '../.';


describe('createAuthConfig(environmentObject)', () => {
  describe('when PASSWORD_SALT_ROUNDS was defined properly', () => {
    it('should return correct config', () => {
      const environmentObject = {
        PASSWORD_SALT_ROUNDS: 10,
      };

      const result = createAuthConfig(environmentObject);

      expect(result).toMatchSnapshot();
    });
  });

  describe('when PASSWORD_SALT_ROUNDS was not defined', () => {
    it('should throw', () => {
      const environmentObject = {};

      expect(createAuthConfig.bind(null, environmentObject)).toThrowError(/Auth config validation error/);
    });
  });

  describe('when PASSWORD_SALT_ROUNDS is not a number', () => {
    it('should throw', () => {
      const environmentObject = {
        PASSWORD_SALT_ROUNDS: 'some string',
      };

      expect(createAuthConfig.bind(null, environmentObject)).toThrowError(/Auth config validation error/);
    });
  });
});
