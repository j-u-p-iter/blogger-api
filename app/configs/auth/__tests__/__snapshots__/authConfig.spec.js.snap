// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createAuthConfig(environmentObject) when PASSWORD_SALT_ROUNDS was defined properly should return correct config 1`] = `
Object {
  "PASSWORD_SALT_ROUNDS": 10,
}
`;
